scanpipe/pipes/d2d.py

```diff
@@ -817,26 +822,119 @@ def _map_javascript_path_resource(
         if not match or len(match.resource_ids) > match.matched_path_length:
             continue
 
+        # Don't map resources solely based on their names.
+        if match.matched_path_length <= 1:
+            continue
+
         if match.matched_path_length > max_matched_path:
             max_matched_path = match.matched_path_length
             from_resource = from_resources.get(id=match.resource_ids[0])
             extra_data = {"path_score": f"{match.matched_path_length}/{path_parts_len}"}
 
-    if not from_resource:
-        return 0
+    return js.map_related_files(
+        to_resources,
+        to_resource,
+        from_resource,
+        map_type,
+        extra_data,
+    )
 
-    transpiled = [to_resource]
-    for related_ext in prospect.get("related", []):
-        try:
-            transpiled.append(to_resources.get(path=f"{base_path}{related_ext}"))
-        except CodebaseResource.DoesNotExist:
-            pass
 
-    for match in transpiled:
-        pipes.make_relation(
-            from_resource=from_resource,
-            to_resource=match,
-            map_type="js_path",
-            extra_data=extra_data,
+def map_javascript_colocation(project, logger=None):
+    """Map JavaScript files based on neighborhood file mapping."""
+    project_files = project.codebaseresources.files().only("path")
```

Comment:
On closer look, it seems like we're better off not using `only("path")`.
Getting `path` quickly won't help much, since in successive lines, we also need `tag`, `extension`, `status`, and `name` to get the desired list of to_resources.

Reply:
Using `only` can be tricky and trigger more queries than not using it. Especially in code that is evolving fast like the d2d pipes.